public.tf

```diff
@@ -83,29 +118,51 @@ resource "aws_eip" "public" {
 }
 
 resource "aws_nat_gateway" "public" {
-  count         = "${local.public_nat_gateways_count}"
-  allocation_id = "${element(aws_eip.public.*.id, count.index)}"
-  subnet_id     = "${element(aws_subnet.public.*.id, count.index)}"
-  depends_on    = ["aws_subnet.public"]
+  count         = local.public_nat_gateways_count
+  allocation_id = element(aws_eip.public.*.id, count.index)
+  subnet_id     = element(aws_subnet.public.*.id, count.index)
+  depends_on    = [aws_subnet.public]
 
   lifecycle {
     create_before_destroy = true
   }
 
-  tags = "${
-    merge(
-      module.public_label.tags,
-      map(
-        "Name", "${module.public_label.id}${var.delimiter}${element(var.availability_zones, count.index)}",
-        "AZ", "${element(var.availability_zones, count.index)}",
-        "Type", "${var.type}"
-      )
-    )
-  }"
+  tags = merge(
+    module.public_label.tags,
+    {
+      "Name" = "${module.public_label.id}${var.delimiter}${element(var.availability_zones, count.index)}"
+      "AZ"   = element(var.availability_zones, count.index)
+      "Type" = var.type
+    },
+  )
 }
 
 # Dummy list of NAT Gateway IDs to use in the outputs for private subnets and when `nat_gateway_enabled=false` for public subnets
 # Needed due to Terraform limitation of not allowing using conditionals with maps and lists
 locals {
-  dummy_az_ngw_ids = ["${slice(list("0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0"), 0, length(var.availability_zones))}"]
+  dummy_az_ngw_ids = [slice(
```

Comment:
This should be just `slice(...)`. Wrapping in list results in nested list such as: `[["0", "0"]]` and causes error in zipmap:

```
  18:   value = zipmap(
  19:
  20:
  21:
    |----------------
    | aws_nat_gateway.public is empty tuple
    | local.dummy_az_ngw_ids is tuple with 1 element
    | var.availability_zones is list of string with 3 elements
```